### ISSUE TITLE
Requiring sudo privilages

### ISSUE THREAD

**Issue by mluscon** [apache/airflow#21487]
### Apache Airflow version

2.2.3 (latest released)

### What happened

If airflow worker process is run under non-root user and a dag contains run_as statement, airflow calls `sudo chown` during the creation of tmp files. The following code snippet comes from /[airflow](https://github.com/apache/airflow/tree/main/airflow)/[task](https://github.com/apache/airflow/tree/main/airflow/task)/[task_runner](https://github.com/apache/airflow/tree/main/airflow/task/task_runner)/base_task_runner.py .
```
        if self.run_as_user and (self.run_as_user != getuser()):
            # We want to include any environment variables now, as we won't
            # want to have to specify them in the sudo call - they would show
            # up in `ps` that way! And run commands now, as the other user
            # might not be able to run the cmds to get credentials
            cfg_path = tmp_configuration_copy(chmod=0o600, include_env=True, include_cmds=True)

            # Give ownership of file to user; only they can read and write
            subprocess.check_call(
                ['sudo', 'chown', self.run_as_user, cfg_path, self._error_file.name], close_fds=True
            )

            # propagate PYTHONPATH environment variable
            pythonpath_value = os.environ.get(PYTHONPATH_VAR, '')
            popen_prepend = ['sudo', '-E', '-H', '-u', self.run_as_user]
```
Calling `sudo chown` in subprocess actually requires airflow process user to be able to sudo to root which averts the main point of running airflow worker process as a non-root user. I would therefore propose to allow run_as_user read and write by setting facl attributes of tmp file accordingly. 

Will you be willing to accept a PR addressing this issue via setfacl?

### Operating System

Linux

### Deployment

Docker-Compose

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
Looked a bit closer - I assume you only want to add the "airflow" user a capability to only sudo to "run_as_user" but not to root? 
Seems doable.

Did you test if all will work in this case? Does FACL not require any special permissions to set the file to be owner by the "run_as_user""? 

If  so - feel free to make a PR. But please remember to update "impersonation" documentation.

**Comment by potiuk** [comment]
Ah yeah - `acl` needs to be enabled when you mount partition, so while it does make sense, we need to add it to impersonation prerequisites and provide fallback to sudo in this case.

**Comment by potiuk** [comment]
But it is a nice feature indeed.

BTW. Just to correct your statement - the current option with sudo does NOT avert the main purpose. Maybe there is a misunderstanding about the purpose of impersonation.

The purpose of impersonation is not to prevent the "airflow" user to run as root, but to make sure that the "task" coming from DAG wirtten by DAG author is executed only using the "target user" (which does not have to use sudo). So it worked fine before as expected. 

The FACL solution simply changes the requirements that the airflow user must fullfill to be able to use. Instead of having `sudo to root` the "${TMP_DIR}"  folder must be mounted on acl-enabled volume and the airlfow user should be able to `sudo as run_for_user`.

**Comment by mluscon** [comment]
> Looked a bit closer - I assume you only want to add the "airflow" user a capability to only sudo to "run_as_user" but not to root? Seems doable.

Exactly, my "airflow" user is capable of impersonating all of run_as_users that I am using but not root.

**Comment by eladkal** [comment]
@mluscon will you PR a fix?

**Comment by G-S-Paris** [comment]
@mluscon, @potiuk 
I have this same usecase - I want to use airflow and execute tasks as particular users; I can even assign my "airflow-user" sudoer access to the "particular users", but giving our airflow-user root access is off the table unfortunately.
Where did we get with the change previously? I haven't worked with FACL before but I'm willing to give it a shot

**Comment by potiuk** [comment]
Have you seen https://airflow.apache.org/docs/apache-airflow/1.10.10/security.html?highlight=impersonation#impersonation ? @G-S-Paris - maybe that will be enough for your case.

**Comment by potiuk** [comment]
BTW. I guess - if not - then no-nothing happened here as far as I see (and I see the same that you see) and if you would like to pick it up, then feel free @G-S-Paris
